src/pk_core.c

```diff
@@ -912,6 +912,71 @@ DEF(stdMathTangent,
   RET(VAR_NUM(tan(rad)));
 }
 
+DEF(stdMathSinh,
+  "sinh(val) -> val\n"
+  "Return the hyperbolic sine value of the argument [val].") {
+
+  double val;
+  if (!validateNumeric(vm, ARG(1), &val, "Argument 1")) return;
+  RET(VAR_NUM(sinh(val)));
+}
+
+DEF(stdMathCosh,
+  "cosh(val) -> val\n"
+  "Return the hyperbolic cosine value of the argument [val].") {
+
+  double val;
+  if (!validateNumeric(vm, ARG(1), &val, "Argument 1")) return;
+  RET(VAR_NUM(cosh(val)));
+}
+
+DEF(stdMathTanh,
+  "tanh(val) -> val\n"
+  "Return the hyperbolic tangent value of the argument [val].") {
+
+  double val;
+  if (!validateNumeric(vm, ARG(1), &val, "Argument 1")) return;
+  RET(VAR_NUM(tanh(val)));
+}
+
+DEF(stdMathArcSine,
+  "asin(num) -> num\n"
+  "Return the arcsine value of the argument [num] which is an angle "
+  "expressed in radians.") {
+
+  double num;
+  if (num < -1 && num > 1)
+    RET_ERR(newString(vm, "Argument should be between -1 and 1"));
+
+  if (!validateNumeric(vm, ARG(1), &num, "Argument 1")) return;
+
+  RET(VAR_NUM(asin(num)));
+}
+
+DEF(stdMathArcCosine,
+  "acos(num) -> num\n"
+  "Return the arc cosine value of the argument [num] which is "
+  "an angle expressed in radians.") {
+
+  double num;
+  if (num < -1 && num > 1)
+    RET_ERR(newString(vm, "Argument should be between -1 and 1"));
+
+  if (!validateNumeric(vm, ARG(1), &num, "Argument 1")) return;
+
+  RET(VAR_NUM(acos(num)));
```

Comment:
1. the `num` here is uninitialized and you're trying to check its value range.
2. First get the num from the stack using `if (!validateNumeric(...` this will set the `num` value (because we're passing the num as a pointer then, the function will update its value), and once after that check the range.
3. `if (num < -1 && num > 1)` this will always false this should be `(num < -1 || 1 < num)`
4. I'd like the error message to be `"... between -1 and +1"` (explicitly mark the positive sign)
5. Same for `asin()`